backstory/src/main/java/com/google/sps/servlets/GetBackstoryServlet.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Text;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.sps.servlets.data.Backstory;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet which gets the Backstory resource. Currently, only the most recently uploaded Backstory
+ * will be gathered from permanent storage then sent back to the front-end by writing it to the
+ * response.
+ */
+@WebServlet("/backstory")
+public class GetBackstoryServlet extends HttpServlet {
```

Comment:
What's the difference between this servlet and the GetAnalyzedImages servlet? I'm wondering if some of the duplication can be removed by consolidating these into a single servlet that accepts a query argument.

Reply:
It is necessary to have to servlets, as far as I know, because of the blobstore service fetch required to serve back the image. Blobstore fetch only returns the BLOB in the request.

Reply:
But in order to avoid code duplication, I put the logic to get a query into a public function in blobstore manager, which both servlets can use.